clang/lib/Parse/ParseExpr.cpp

```diff
@@ -221,6 +221,15 @@ ExprResult Parser::ParseConstantExpression() {
   return ParseConstantExpressionInExprEvalContext(NotTypeCast);
 }
 
+ExprResult Parser::ParseArrayBoundExpression() {
+  EnterExpressionEvaluationContext ConstantEvaluated(
+      Actions, Sema::ExpressionEvaluationContext::ConstantEvaluated);
+  // If we parse the bound of a VLA... we parse a non-constant
+  // constant-expression!
+  Actions.ExprEvalContexts.back().InConditionallyConstantEvaluateContext = true;
```

Comment:
I think I'd rather this be an argument to the above constructor here.

Reply:
@erichkeane this constructor already has a bunch of defaulted boolean / pointer parameters, it would be a mess. We'd need to add a tag which seems overkill for 1 use

Reply:
In reality, I'd probably prefer something like; 

``` auto Expr = ExpressionEvaluationContext::CreateForVLA```

kind of thing (or perhaps a builder-pattern like below), but that is probably a sizable refactor for everything.  I think I can hold my nose here and be OK for now.

``` ExprEvalContext Eva(ThingsEverythingNeeds).ForVLA()```